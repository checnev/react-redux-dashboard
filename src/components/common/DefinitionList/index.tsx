import React, { Fragment } from 'react';
import {
  Typography,
  Grid,
} from '@material-ui/core';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

interface Definition {
  name: string;
  value: React.ReactNode;
}

interface DefinitionListProps {
  data: Definition[];
  nameVariant?: 'h4' | 'h5' | 'h6' | 'body1' | 'body2';
  valueVariant?: 'h4' | 'h5' | 'h6' | 'body1' | 'body2';
}

const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {
    marginTop: 0,
    marginBottom: 0,
  },
  ceil: {
    borderBottom: '1px solid rgba(0, 0, 0, 0.12)',
  }
}));

const DefinitionList: React.FC<DefinitionListProps> = ({
  data,
  nameVariant,
  valueVariant,
}: DefinitionListProps) => {
  const classes = useStyles();
  return (
    <Grid 
      container
      spacing={2}
      component="dl"
      className={classes.root}
    >
      {data.map((item) => (
        <Fragment key={item.name}>
          <Grid item xs={6} className={classes.ceil}>
            <Typography 
              component="dt" 
              variant={nameVariant || 'body1'}
            >
              {item.name}:
            </Typography>
          </Grid>
          <Grid item xs={6} className={classes.ceil}>
            <Typography 
              component="dd"
              variant={valueVariant || 'body1'}
            >
              {item.value}
            </Typography>
          </Grid>
        </Fragment>
      ))}
    </Grid>
  );
};

export default DefinitionList;
