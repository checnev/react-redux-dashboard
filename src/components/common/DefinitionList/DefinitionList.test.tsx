import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from 'pretty';
import DefinitionList from 'components/common/DefinitionList';

let container: any = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});


it('DefinitionList render', () => {
  const data = [
    {
      name: 'Name 1',
      value: 'Value 1'
    },
    {
      name: 'Name 2',
      value: 'Value 2',
    }
  ];

  act(() => {
    render(<DefinitionList data={data} />, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
