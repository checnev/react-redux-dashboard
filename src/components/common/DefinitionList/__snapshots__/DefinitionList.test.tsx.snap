// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefinitionList render 1`] = `
"<dl class=\\"MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2\\">
  <div class=\\"MuiGrid-root makeStyles-ceil-2 MuiGrid-item MuiGrid-grid-xs-6\\">
    <dt class=\\"MuiTypography-root MuiTypography-body1\\">Name 1:</dt>
  </div>
  <div class=\\"MuiGrid-root makeStyles-ceil-2 MuiGrid-item MuiGrid-grid-xs-6\\">
    <dd class=\\"MuiTypography-root MuiTypography-body1\\">Value 1</dd>
  </div>
  <div class=\\"MuiGrid-root makeStyles-ceil-2 MuiGrid-item MuiGrid-grid-xs-6\\">
    <dt class=\\"MuiTypography-root MuiTypography-body1\\">Name 2:</dt>
  </div>
  <div class=\\"MuiGrid-root makeStyles-ceil-2 MuiGrid-item MuiGrid-grid-xs-6\\">
    <dd class=\\"MuiTypography-root MuiTypography-body1\\">Value 2</dd>
  </div>
</dl>"
`;
